src/node.js

```diff
@@ -10,187 +10,218 @@ const shutdown = require('shutdown')
 const path = require('path')
 const join = path.join
 
-const rootPath = process.env.testpath ? process.env.testpath : __dirname
-
-const ipfsDefaultPath = findIpfsExecutable()
+const BinWrapper = require('bin-wrapper')
+
+// TODO maybe this should go to some config file
+const DIST_BASE = 'https://dist.ipfs.io/go-ipfs'
+const VERSION = 'v0.4.1'
+const PROJECT_BASE = DIST_BASE + '/' + VERSION + '/go-ipfs_' + VERSION + '_'
+const IPFS_DEFAULT_EXEC_PATH = path.join(__dirname, '..', 'vendor')
+
+const bin = new BinWrapper()
+    .src(PROJECT_BASE + 'darwin-386.tar.gz', 'darwin', 'ia32')
+    .src(PROJECT_BASE + 'darwin-amd64.tar.gz', 'darwin', 'x64')
+    .src(PROJECT_BASE + 'freebsd-amd64.tar.gz', 'freebsd', 'x64')
+    .src(PROJECT_BASE + 'linux-386.tar.gz', 'linux', 'ia32')
+    .src(PROJECT_BASE + 'linux-amd64.tar.gz', 'linux', 'x64')
+    .src(PROJECT_BASE + 'linux-arm.tar.gz', 'linux', 'arm')
+    .src(PROJECT_BASE + 'windows-386.tar.gz', 'win32', 'ia32')
+    .src(PROJECT_BASE + 'windows-amd64.tar.gz', 'win32', 'x64')
+    .use(process.platform === 'win32' ? 'ipfs.exe' : 'ipfs')
+    // .version('>=1.71');
 
 const GRACE_PERIOD = 7500 // amount of ms to wait before sigkill
 
-function findIpfsExecutable () {
-  let npm3Path = path.join(rootPath, '../../', '/go-ipfs-dep/go-ipfs/ipfs')
-
-  let npm2Path = path.join(rootPath, '../', 'node_modules/go-ipfs-dep/go-ipfs/ipfs')
-
-  try {
-    fs.statSync(npm3Path)
-    return npm3Path
-  } catch (e) {
-    return npm2Path
-  }
-}
-
 function configureNode (node, conf, done) {
   const keys = Object.keys(conf)
   series(keys.map((key) => (cb) => {
     const value = conf[key]
     const env = {env: node.env}
 
-    run(node.exec, ['config', key, '--json', JSON.stringify(value)], env)
-      .on('error', cb)
-      .on('end', cb)
+    node._run(['config', key, '--json', JSON.stringify(value)], env, cb)
   }), done)
 }
 
-// Consistent error handling
 function parseConfig (path, done) {
-  try {
-    const file = fs.readFileSync(join(path, 'config'))
-    const parsed = JSON.parse(file)
-    done(null, parsed)
-  } catch (err) {
-    done(err)
-  }
+  fs.readFile(join(path, 'config'), (err, data) => {
+    if (err) return done(err)
+    // Consistent error handling
+    try {
+      const parsed = JSON.parse(data)
+      return done(null, parsed)
+    } catch (err) {
+      return done(err)
+    }
+  })
```

Comment:
While I was debugging why 0.4.2 didn't work ended up changing this and didn't change it back. But any particular reason why was this sync?